torch/_functorch/_aot_autograd/traced_function_transforms.py

```diff
@@ -350,12 +349,28 @@ def _functionalized_f_helper(*args):
         disable_above = torch._C._ExcludeDispatchKeyGuard(
             torch._C.DispatchKeySet(torch._C.DispatchKey.Functionalize)
         )
-        with disable_above, FunctionalTensorMode(aot_config.pre_dispatch):
+
+        with disable_above:
             # Wrap inputs into functional wrappers
             f_args = pytree.tree_map(to_fun, args)
+
+            # Populate the current FunctionalTensorMode with the tokens per
+            # operator
+            tokens = f_args[: len(meta.tokens)]
```

Comment:
One sanity assert you could do here is to assert that `[x.dim == 0 for x in tokens]` (a common "failure mode" in aot autograd historically is that we mess up our indexing somewhere and plumb the wrong tensors around)